gh-127833: Add links to token types to the lexical analysis intro (#131468)
python/cpython: Doc/reference/lexical_analysis.rst

Except at the beginning of a logical line or in string literals, the whitespace
characters space, tab and formfeed can be used interchangeably to separate
tokens.  Whitespace is needed between two tokens only if their concatenation
could otherwise be interpreted as a different token. For example, ``ab`` is one

> token, but ``a b`` is two tokens. For another example, ``+a`` is two tokens,
> and is equivalent to ``+ a``.

Reviewer: I'm not keen on 'equivalent to', perhaps:
```suggestion
token, but ``a b`` is two tokens. However, ``+a`` and ``+ a`` both tokenize to the same output, as ``+a`` is not a valid token.
```

Author: Maybe like this?
```suggestion
token, but ``a b`` is two tokens. However, ``+a`` and ``+ a`` both produce
two tokens, ``a`` and ``b``, as ``+a`` is not a valid token.
```

> token, but ``a b`` is two tokens. However, ``+a`` and ``+ a`` both produce
> two tokens, ``a`` and ``b``, as ``+a`` is not a valid token.

Reviewer: Thanks for the suggested new wording, but I think there's an error:
```suggestion
token, but ``a b`` is two tokens. However, ``+a`` and ``+ a`` both produce
two tokens, ``+`` and ``a``, as ``+a`` is not a valid token.
```

Author: :man_facepalming:

Author: Thanks for the catch!
